### PR TITLE
Add editor and console hats

### LINKED ISSUE #4223: Console + editor onboarding flow
Step two of [Project Mad Hatter](#4214)

Resources:
[Discord channel](https://discord.com/channels/779097926135054346/903353850566672465/903353852814839849) 
[Jason's first Loom](https://www.loom.com/share/ea7bf4e14fc845f2bb2998bd5b1c3b08) 
[Figma](https://www.figma.com/file/SSr1ljyzF0lCXTL7BX1Whh/?node-id=7995%3A50177) 

**Console**
The key with this visual is that it's using the exact same spacing as is seen in #4222. This screenshot is in Figma, not taking a screenshot of the actual UI, so things won't line up exactly right. But the idea is that #4222 and #4223 should match. (Copy may change but of course that's easy)

image

**Editor**
Guess what?! SAME DESIGN, BY DESIGN.

image

Everything should be identical because we're using a single hat component.

**Who should see this?**
More long term, we're going to make these nags dynamic. For example, if it's your own Replay, we might use this copy. But if it's a publicly shared Replay, we might consider hats with different copy. But that will come later.

**For now, we'll show nags to:**
* The author of the Replay
* If they've never seen the nags before

**FAQ:**
Q: Is it possible for someone to see the console nag at the same time as the editor nag? For example if the editor window has been opened?
A: My proposal is that we don't allow that. Only show the editor nag once the console nag has been dismissed because someone has clicked in the console.

### PR DESCRIPTION
**Pull request by jaril**
Fix #4223.

image

### CODE REVIEW

File: src/ui/components/shared/Nags/Nags.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    </div>
+  );
+}
+
+export function EditorNag() {
+  const { nags } = hooks.useGetUserInfo();
+
+  // Don't show the editor nag until the user has gotten past the
+  // console navigate nag.
+  if (!nags.includes(Nag.FIRST_CONSOLE_NAVIGATE)) {
```
Comment (jasonLaster) [inline src/ui/components/shared/Nags/Nags.tsx:38]: @jazzdan @ryanjduffy is it easy/possible to set up a launch darkly A/B test for this.

I'd love to say half of our users should see this NAG regardless of whether they've first clicked on console navigate.

Comment (ryanjduffy) [reply src/ui/components/shared/Nags/Nags.tsx:38]: should be. I added support for LD in devtools before launch so it should be a bit of work on the LD side and a bit in code but nothing too bad

Comment (jasonLaster) [reply src/ui/components/shared/Nags/Nags.tsx:38]: Great!

https://github.com/RecordReplay/devtools/blob/a0c76b5d2ee4265f8eb7c3f87d106770cadca546/src/ui/components/shared/WorkspaceSettingsModal/BillingBanners.tsx#L3-L4

File: src/ui/components/shared/Nags/Nags.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    />
+  );
+}
+
+export function ConsoleNag() {
+  return (
+    <NagHat
+      mainText="Check out our best feature"
+      subText="Try playing to one of our console logs"
+      nagType={Nag.FIRST_CONSOLE_NAVIGATE}
```
Comment (jasonLaster) [inline src/ui/components/shared/Nags/Nags.tsx:67]: ## Want to see something cool?
### Try fast forwarding or rewinding to a console log.

File: src/ui/components/shared/Nags/Nags.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  // console navigate nag.
+  if (!nags.includes(Nag.FIRST_CONSOLE_NAVIGATE)) {
+    return null;
+  }
+
+  return (
+    <NagHat
+      mainText="Ready to add your first print statement?"
+      subText="Click on a line number in the gutter"
+      nagType={Nag.FIRST_GUTTER_CLICK}
```
Comment (jasonLaster) [inline src/ui/components/shared/Nags/Nags.tsx:57]: 👍

**Review by jasonLaster — APPROVED** [review]
small tweak to the copy

File: src/devtools/client/webconsole/components/Output/Message.js [hunk 4]
```diff
@@ -181,6 +182,7 @@ class Message extends Component {
       );
 
       this.onViewSourceInDebugger({ ...frame, url: frame.source });
+      dismissNag(Nag.FIRST_CONSOLE_NAVIGATE);
```
Comment (jasonLaster) [inline src/devtools/client/webconsole/components/Output/Message.js:185]: hmm - the nag didnt go away for me when i clicked fast forward. I think this needs to be in the fastforward click handler...

**Review by jasonLaster — APPROVED** [review]

**Pull request merged**